src/Data/UriPath.hs

```diff
@@ -33,16 +33,20 @@ instance IsString UriPart where
 instance s ~ ST => ConvertibleStrings UriPart s where
     convertString = unUriPart
 
--- | An @UriPath@ is a list of @UriPart@s stored as a difference list.
+-- | An @UriPath@ is (1) a list of @UriPart@s stored as a difference list, and (2) an assoc list
+-- containing the query.
+--
+-- When concatenating queries in 'Monoid', the query params of the parts are collected at the end of
+-- the path.  Matrix parameters would offer a way to keep query params near the path segments they
+-- aim at.  Mentioned in RFC3986 as "has been witnessed in the wild, but not often".
```

Comment:
In this version of [rfc3986](https://www.ietf.org/rfc/rfc3986.txt) I couldn't find the sentence, can you give us the link to the version you used?